components/conversion-form.tsx

```diff
@@ -55,50 +58,56 @@ export const ConversionForm = ({
-        <Input
-          autoFocus
-          id={`${type}-input`}
-          type="number"
-          value={input}
-          onChange={handleInputChange}
-        />
-      </div>
-      <div className="grid grid-cols-2 gap-4">
+    <>
+      <div className="space-y-4">
         <div className="space-y-2">
-          <Label htmlFor={`${type}-from`}>From</Label>
-          <Select value={fromUnit} onValueChange={handleFromUnitChange}>
-            <SelectTrigger id={`${type}-from`}>
-              <SelectValue placeholder="Select unit" />
-            </SelectTrigger>
-            <SelectContent>
-              <UnitDropdownOptions units={units} />
-            </SelectContent>
-          </Select>
+          <Label htmlFor={`${type}-input`}>Value</Label>
+          <Input
+            autoFocus
+            id={`${type}-input`}
+            type="number"
+            value={input}
+            onChange={handleInputChange}
+          />
+        </div>
+        <div className="grid grid-cols-2 gap-4">
+          <div className="space-y-2">
+            <Label htmlFor={`${type}-from`}>From</Label>
+            <Select value={fromUnit} onValueChange={handleFromUnitChange}>
+              <SelectTrigger id={`${type}-from`}>
+                <SelectValue placeholder="Select unit" />
+              </SelectTrigger>
+              <SelectContent>
+                <UnitDropdownOptions units={units} />
+              </SelectContent>
+            </Select>
+          </div>
+          <div className="space-y-2">
+            <Label htmlFor={`${type}-to`}>To</Label>
+            <Select value={toUnit} onValueChange={handleToUnitChange}>
+              <SelectTrigger id={`${type}-to`}>
+                <SelectValue placeholder="Select unit" />
+              </SelectTrigger>
+              <SelectContent>
+                <UnitDropdownOptions units={units} />
+              </SelectContent>
+            </Select>
+          </div>
         </div>
         <div className="space-y-2">
-          <Label htmlFor={`${type}-to`}>To</Label>
-          <Select value={toUnit} onValueChange={handleToUnitChange}>
-            <SelectTrigger id={`${type}-to`}>
-              <SelectValue placeholder="Select unit" />
-            </SelectTrigger>
-            <SelectContent>
-              <UnitDropdownOptions units={units} />
-            </SelectContent>
-          </Select>
+          <Label htmlFor={`${type}-result`}>Result</Label>
+          <Input
+            id={`${type}-result`}
+            type="text"
+            value={formattedResult}
+            readOnly
+          />
         </div>
       </div>
-      <div className="space-y-2">
-        <Label htmlFor={`${type}-result`}>Result</Label>
-        <Input
-          id={`${type}-result`}
-          type="text"
-          value={formattedResult}
-          readOnly
-        />
-      </div>
-    </div>
+
+      {type === ConversionType.Volume && <VolumeTable />}
```

Comment:
O I have one more request please, we can get rid of these conditionals by just passing the component as a prop. Something like 

```
 <ConversionForm
              type={ConversionType.Temperature}
              units={temperatureUnits}
              input={input}
              setInput={setInput}
              convert={convertTemperature}
              convertionTable= {<TemperatureTable /
>}
            />
            ```